File: iwsoj/runner/alg_languages/prime_numbers.go
package main

import (
	"fmt"
	"math"
	"os"
	"bufio"
	"log"
	"strconv"
)


func prime(n int) int {
	if n < 2 {
		return 0
	} else {
		counter := 0
		for i := 2; i <= n; i++ {
			counter = counter + 1
			for j := 2; j <= (int(math.Round(math.Sqrt(float64(i))))); j++ {
				if i % j == 0 {
					counter = counter - 1
					break
				}
			} 
		}
		return counter
	}
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		i1, err := strconv.Atoi(scanner.Text())
		if err == nil {
			fmt.Println(prime(i1))
		}
	}

	if err := scanner.Err(); err != nil {
		log.Println(err)
	}
}

File: iwsoj/runner/alg_languages/prime_numbers.py
import sys
import math

def prime(n):
    if n < 2:
        return 0;
    else:
        counter = 0
        for i in range(2, (n+1)):
            counter = counter + 1
            for j in range(2, (math.floor(math.sqrt(i))+1)):
                if i % j == 0:
                    counter = counter - 1
                    break
        return counter

                
theInts = []
i = 0
for val in sys.stdin.read().split():
    theInts.append(int(val))
    print(prime(theInts[i]))
    i = i + 1